cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestBase.java

```diff
@@ -156,9 +160,24 @@ public Boolean call() throws Exception {
         if (!getMonitorClient().allSystemServicesOk()) {
           return false;
         }
-        // Check that the dataset service is up with list(). If list() does not throw exception, which means the http
-        // request receives response status HTTP_OK and dataset service is up, return true.
-        getNamespaceClient().list();
+
+        // Wait for the default namespace to exist otherwise there can be a race condition where namespace is being
+        // created but the integration test has started
+        Tasks.waitFor(true, new Callable<Boolean>() {
+          @Override
+          public Boolean call() throws Exception {
+            // Check that the dataset service is up with list(). If list() does not throw exception, which means the
+            // http request receives response status HTTP_OK and dataset service is up, return true.
+            List<NamespaceMeta> list = getNamespaceClient().list();
+            for (NamespaceMeta namespaceMeta : list) {
```

Comment:
this check should only be done if the configured namespace is the default namespace.